src/Aspire.Hosting.Qdrant/QdrantBuilderExtensions.cs

```diff
@@ -0,0 +1,97 @@
+
+                // If in Publish mode, disable static content, which disables the Dashboard Web UI
+                // https://github.com/qdrant/qdrant/blob/acb04d5f0d22b46a756b31c0fc507336a0451c15/src/settings.rs#L36-L40
+                if (builder.ExecutionContext.IsPublishMode)
+                {
+                    context.EnvironmentVariables[DisableStaticContentEnvVarName] = "0";
+                }
+            });
+    }
+
+    /// <summary>
+    /// Adds a named volume for the data folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the resource name.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only volume.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataVolume(this IResourceBuilder<QdrantServerResource> builder, string? name = null, bool isReadOnly = false)
+        => builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataBindMount(this IResourceBuilder<QdrantServerResource> builder, string source, bool isReadOnly = false)
+        => builder.WithBindMount(source, "/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Add a reference to a Qdrant settings for a project.
+    /// </summary>
+    /// <param name="builder">An <see cref="IResourceBuilder{T}"/> for <see cref="ProjectResource"/></param>
+    /// <param name="qdrantResource">The Qdrant server resource</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ProjectResource> WithReference(this IResourceBuilder<ProjectResource> builder, IResourceBuilder<QdrantServerResource> qdrantResource)
+    {
+        builder.WithEnvironment(context =>
+        {
+            // primary endpoint (gRPC)
+            context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}"] = new ConnectionStringReference(qdrantResource.Resource, optional: false);
+
+            // REST endpoint
+            var restEndpoint = qdrantResource.Resource.GetEndpoint("rest");
+            context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}_rest"] = ReferenceExpression.Create($"Endpoint={restEndpoint};Key={qdrantResource.Resource.ApiKeyParameter}");
```

Comment:
One way to make this a bit cleaner is to make it a property on the resource and just use it here.

Reply:
Like this?

Resource:
    public ReferenceExpression RestConnectionStringExpression =>
        ReferenceExpression.Create(
            $"Endpoint={PrimaryEndpoint.Property(EndpointProperty.Scheme)}://{PrimaryEndpoint.Property(EndpointProperty.Host)}:{PrimaryEndpoint.Property(EndpointProperty.Port)}_rest;Key={ApiKeyParameter}");

WithReference overload:
// primary endpoint (gRPC)
context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}"] = qdrantResource.Resource.ConnectionStringExpression;

// REST endpoint
context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}_rest"] = qdrantResource.Resource.RestConnectionStringExpression;